src/pydvl/value/sampler.py

```diff
@@ -142,17 +144,10 @@ def complement(self, exclude: Sequence[T]) -> NDArray[T]:
         return np.setxor1d(self._indices, exclude)
 
     def iterindices(self) -> Iterator[T]:
-        """Iterates over indices in the order specified at construction.
-
-        FIXME: this is probably not very useful, but I couldn't decide
-          which method is better
-        """
+        """Iterates over indices in the order specified at construction."""
         if self._index_iteration is PowersetSampler.IndexIteration.Sequential:
             for idx in self._outer_indices:
                 yield idx
-        elif self._index_iteration is PowersetSampler.IndexIteration.Random:
```

Comment:
You just removed the else case but it still exists and is not handled.

Reply:
Readded.